Go/main.go

```diff
@@ -0,0 +1,489 @@
+	if err != nil {
+		log.Fatalf("failed to create cosmos db account client: %v", err)
+	}
+
+	if _, err := accountClient.Get(ctx, resourceGroupName, accountName, nil); err != nil {
+		log.Fatalf("failed to get cosmos db account: %v", err)
+	}
+
+	pollerResp, err := databaseClient.BeginCreateUpdateSQLDatabase(ctx, resourceGroupName, accountName, databaseName, properties, nil)
+	if err != nil {
+		log.Fatalf("failed to begin create or update cosmos db database: %v", err)
+	}
+
+	resp, err := pollerResp.PollUntilDone(ctx, nil)
+	if err != nil {
+		log.Fatalf("failed to poll the result: %v", err)
+	}
+
+	fmt.Printf("Created new Database: %s\n", *resp.ID)
+}
+
+func createOrUpdateCosmosDBContainer(ctx context.Context) {
+	containerClient, err := armcosmos.NewSQLResourcesClient(subscriptionID, credential, nil)
+	if err != nil {
+		log.Fatalf("failed to create cosmos db container client: %v", err)
+	}
+
+	partitionKind := armcosmos.PartitionKindMultiHash
+	indexingMode := armcosmos.IndexingModeConsistent
+	conflictResolutionModeLastWriterWins := armcosmos.ConflictResolutionModeLastWriterWins
+
+	properties := armcosmos.SQLContainerCreateUpdateParameters{
+		Location: &location,
+		Properties: &armcosmos.SQLContainerCreateUpdateProperties{
+			Resource: &armcosmos.SQLContainerResource{
+				ID: &containerName,
+				PartitionKey: &armcosmos.ContainerPartitionKey{
+					Paths:   []*string{to.StringPtr("/companyId"), to.StringPtr("/departmentId"), to.StringPtr("/userId")},
+					Kind:    &partitionKind,
+					Version: to.Int32Ptr(2),
+				},
+				IndexingPolicy: &armcosmos.IndexingPolicy{
+					Automatic:    to.BoolPtr(true),
+					IndexingMode: &indexingMode,
+					IncludedPaths: []*armcosmos.IncludedPath{
+						{Path: to.StringPtr("/*")},
+					},
+					ExcludedPaths: []*armcosmos.ExcludedPath{
+						{Path: to.StringPtr("/\"_etag\"/?")},
+					},
+				},
+				UniqueKeyPolicy: &armcosmos.UniqueKeyPolicy{
+					UniqueKeys: []*armcosmos.UniqueKey{
+						{Paths: []*string{to.StringPtr("/userId")}},
+					},
+				},
+				ConflictResolutionPolicy: &armcosmos.ConflictResolutionPolicy{
+					Mode:                   &conflictResolutionModeLastWriterWins,
+					ConflictResolutionPath: to.StringPtr("/_ts"),
+				},
+			},
+			Options: &armcosmos.CreateUpdateOptions{
+				AutoscaleSettings: &armcosmos.AutoscaleSettings{
+					MaxThroughput: to.Int32Ptr(int32(maxAutoScaleThroughput)),
+				},
+			},
+		},
+	}
+
+	databaseClient, err := armcosmos.NewSQLResourcesClient(subscriptionID, credential, nil)
+	if err != nil {
+		log.Fatalf("failed to create cosmos db database client: %v", err)
+	}
+
+	if _, err := databaseClient.GetSQLDatabase(ctx, resourceGroupName, accountName, databaseName, nil); err != nil {
+		log.Fatalf("failed to get cosmos db database: %v", err)
+	}
+
+	pollerResp, err := containerClient.BeginCreateUpdateSQLContainer(ctx, resourceGroupName, accountName, databaseName, containerName, properties, nil)
```

Comment:
how does this work? shouldn't this be databaseClient making this call? I don't see a reference to containerClient above being created.

Reply:
containerClient, err := armcosmos.NewSQLResourcesClient(subscriptionID, credential, nil) is the first line in the method.